File: covid-api/src/main/java/com/tacs/grupo1/covid/api/dto/statistics/StatisticsDay.java
package com.tacs.grupo1.covid.api.dto.statistics;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.time.OffsetDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
public class StatisticsDay {
    private OffsetDateTime date;
    protected Long confirmed;
    protected Long deaths;
    protected Long recovered;
}

File: covid-frontend/src/models/StatisticsDay.ts
export default class StatisticsDay {
    public date: Date;
    public confirmed: number;
    public deaths: number;
    public recovered: number;

    [key: string]: any;

    constructor(statisticsDay?: any) {
        if (statisticsDay) {
            this.date = statisticsDay.date;
            this.confirmed = statisticsDay.confirmed;
            this.deaths = statisticsDay.deaths;
            this.recovered = statisticsDay.recovered;
        } else {
            this.date = new Date();
            this.confirmed = 0;
            this.deaths = 0;
            this.recovered = 0;
        }
    }
}